code/game/machinery/computer/buildandrepair.dm

```diff
@@ -488,8 +489,13 @@
 				new /obj/item/stack/cable_coil(get_turf(src), 5)
 				return 1
 
-			if(istype(P, /obj/item/stack/sheet/glass/glass))
-				var/obj/item/stack/sheet/glass/glass/G = P
+			if(istype(P, /obj/item/stack/sheet/glass))
+				var/obj/item/stack/sheet/glass/G = P
+				if(istype(G, /obj/item/stack/sheet/glass/rglass) || istype(G, /obj/item/stack/sheet/glass/plasmarglass)) // Don't use these
+					to_chat(user, "<span class='warning'>Sheets of glass must not have rods in them!</span>")
+					return
+				if(istype(G, /obj/item/stack/sheet/glass/plasmaglass)) // For EMP proofing
```

Comment:
You've put this before the amountcheck, meaning you can use 1 plasma glass on the computer to have it cancel AND become emp-proof.

Fix it!

Reply:
Fixed